### ISSUE TITLE
Unreasonable example in documentation

### ISSUE THREAD

**Issue by Stefan-Wagner** [kevinpt/syntrax#9]
On the main page, section "Toploop", there is this example:
  
`toploop(line('(', 'forward', ')'), line(')', 'backward', '('))`

which looks pretty, but it would mean, that after the forward part, with its closing brace, you close another brace, then comes the word "backward", then an opening brace. 

Since the example might be completely made up, this could be valid syntax, but I guess what was meant was: `toploop(line('(', 'forward', ')'), line('(', 'backward', ')'))` which looks awkward in the diagram, because of our habit to read from left to right, but more reasonable.

**Comment by inponomarev** [comment]
Thanks for the issue! It seems that this project is dead.

You may try [jsyntrax](https://github.com/atp-mipt/jsyntrax) instead: this is a reimplementation of `syntrax` in Java.

Your note is taken into account in its [documentation](https://atp-mipt.github.io/jsyntrax/#_toploop).

**Comment by Stefan-Wagner** [comment]
Hey, great to see the project being alive again. (An Easter miracle?) :) 
I used syntrax mainly in Linux, where installing the dependencies wasn't super easy, but manageable. (Not in the Ubuntu-repos up to 20.04). But I teach IT (programming and Linux) to mainly Windows folks, and figuring out how to get syntrax running was a big hassle, so the java dependency is much easier - at least for me, who is already a Java programmer - well, meanwhile a Scala programmer, who needs a JDK although. 

I have 2 feature requests, which I would like seeing discussed. Or 3.
a) Make the title (optional) part of the spec-file. Of course, it is possible to write a script, which takes the title from the filename, but not every useful attribute of a title is fine as filename. We like filenames to be short, but titles might sometimes be more verbose. If I work in an editor like Geany, I can configure it, so that it will produce a PNG when I hit the compile button, but the title has to be set manually, which makes it a less productive workflow. 
b) If I remember correctly, we can put comments into .spec-files, with leading hashes, but that isn't documented. Remarks are very useful, for example to document bugs in the diagram, which can't be fixed immediately. Maybe remarks can be misused, to specify the title (without eliminating the command line option).
c) Sometimes, I would like to have multiple diagrams in one file, where a complex expression can be broken up into subpieces.  That would be a nice feature, too. 

I don't miss the opportunity to build pdf, ps or eps files, because with ImageMagick and other tools, this can be done easily in a pipeline from the SVG or PNG. 

I'm curious to find out, how syntax errors in the spec files are reported with this tool. It was very hard, most of the time, with syntrax, but I guess it's not easy to guess, where the user omitted a closing brace or a comma. :)

**Comment by inponomarev** [comment]
a)  Will you be so kind as to put your first feature-request ("Make the title (optional) part of the spec-file") as a new issue in jsyntrax? We can easily implement this!

b) What was Python in `syntrax` is now Groovy in `jsyntrax`. Thus you can do the following:

```groovy
//comment

/*multi-line 
  comment */

line( '/sign', choice(None, '+', '-') )
```
Unfortunately, 'python-style' comments won't work and should be converted to `//` or `/* */`-style comments.

c) Could you put 'multiple diagrams in one file' as a new issue for jsyntrax? This is also can be easily implemented.

> I'm curious to find out, how syntax errors in the spec files are reported with this tool.

You can try it yourself. Reporting syntax errors is now Groovy interpreter's responsibility, in case of the missed brace it looks like this:

```
>syntrax spec
Something was wrong with input script:
        startup failed:
Script1.groovy: 1: expecting ')', found '' @ line 1, column 39.
   line( '/sign', choice(None, '+', '-')
                                         ^

1 error
```

**Comment by Stefan-Wagner** [comment]
Yes, I can and will do these things. But it might take till the end of the week.

The // - comment possibility I figured out by myself but forgot to report it. Is it documented, already?

**Comment by Stefan-Wagner** [comment]
Finally, I managed to write my feature request - sorry for being so late at this, but I'm very new to this web issues and have to concentrate on this and did not have my brain free, the last weeks. It was much easier than expected. :)

Shall I open a second feature request for multiple graphs, defined in one file? 
Btw.: Is there a convention, how to mark issues as bugs/feature requests? I expected something in the UI, but didn't find it.
